### PR TITLE
[Week1] - seungwoo

### PR DESCRIPTION
**Pull request by Gonzo-Dev-lab**
### 머신러닝
정의: 머신러닝이란 데이터를 사용해 패턴을 학습하고 새로운 데이터에 대한 결과를 예측하는 기술

선형 회귀 모델
선형 회귀는 독립 변수와 종속 변수 간의 직선 관계를 모델링

**k-최근접 이웃 회귀(k-NN)**는 새로운 데이터 포인트와 가장 가까운 k개의 데이터를 사용하여 예측하는 모델

주요 작동 원리:

새로운 데이터와 기존 데이터 간의 거리를 계산
가장 가까운 k개의 데이터를 선택
선택된 데이터들의 평균을 계산하여 예측값으로 사용

### 분류(Classification)
정의: 머신러닝에서 분류는 입력 데이터의 레이블(클래스)을 예측하는 작업입니다.
예: 이메일이 스팸인지 아닌지를 예측하거나, 꽃의 품종을 분류하는 문제.

Logistic Regression, Decision Tree, SVM, Random Forest 

Logistic Regression
분류 알고리즘입니다.
입력 데이터를 분석하여 특정 클래스에 속할 확률을 예측합니다.

### 정확도(Accuracy)
모델이 예측한 값 중에서 실제 레이블과 일치하는 예측의 비율입니다.

과제1.
1. 데이터 준비
아래의 데이터를 사용하여 GDP와 삶의 만족도를 생성하세요.
pandas를 사용해 데이터를 DataFrame으로 변환하세요.
matplotlib를 사용해 데이터를 시각화하세요.
2. 모델 학습
선형 회귀 모델:
**LinearRegression**을 사용해 모델을 정의하고 학습시키세요.
k-최근접 이웃 모델:
**KNeighborsRegressor**를 사용해 ( k=3 )으로 모델을 정의하고 학습시키세요.
3. 예측
키프로스의 1인당 GDP(37655 USD)를 입력하여 두 모델의 삶의 만족도 예측값을 출력하세요.
4. 결과 비교
두 모델의 예측 결과를 비교후, 장단점을 간단히 작성하세요.

과제2.

주어진 데이터에서 꽃의 특성(길이와 너비)을 바탕으로 꽃 품종을 예측하는 모델을 만들어보세요.

1. 데이터 탐색
sklearn.datasets.load_iris를 사용하여 Iris 데이터셋을 불러오세요.
데이터를 간단히 출력하고 특성과 타깃의 형태를 확인하세요.
2. 데이터 분할
데이터를 훈련 세트와 테스트 세트로 나누세요. (train_test_split 사용)
테스트 데이터 비율은 20%로 설정하고, random_state=42로 고정하세요.
3. 모델 정의 및 훈련
LogisticRegression 모델을 정의하고 훈련 데이터로 학습시키세요.
4. 예측 및 평가
테스트 데이터를 사용하여 품종을 예측하세요.
예측 결과를 기반으로 정확도를 출력하세요. (accuracy_score 사용)

### CODE REVIEW

**Review by kimhamyong — COMMENTED** [review]
폴더(week1-seungwoo)에 과제 파일을 모두 넣어서 업로드 부탁드립니다!

task2: 에 대해선 잘 수행하셨습니다.

추가적으로, LogisticRegression에서 solver를 명시적으로 설정하면 데이터셋에 더 최적화된 학습을 수행할 수 있습니다. 특히 작은 데이터셋에서는 'liblinear'가 적합합니다.

**Review by giyomdabin — CHANGES_REQUESTED** [review]
Concept.md 파일 또한, week1-seungwoo 폴더에 넣어서 업로드 부탁드립니다!

task1 : 
주어진 과제에 대한 코드는 잘 동작합니다! 또한, 선형 회귀와 k-최근접 이웃에 대한 특징도 잘 작성하였습니다. 

수고하셨습니다!

**Comment by kimhamyong** [comment]
리드미 파일도 삭제 부탁드려요!
